File: backend/app/models/email.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class EmailBase(BaseModel):
    subject: str
    sender: str
    date: str
    is_unread: bool

class EmailCreate(EmailBase):
    body: str

class EmailResponse(EmailBase):
    id: str
    snippet: Optional[str] = None

class EmailList(BaseModel):
    emails: List[EmailResponse]
    total: int
    unread_count: int

class EmailDetail(EmailResponse):
    body: str
    labels: List[str]

class VoiceCommand(BaseModel):
    command: str

class CommandResponse(BaseModel):
    success: bool
    message: str
    action: Optional[str] = None
    data: Optional[dict] = None 

File: frontend/src/types/email.ts
export interface EmailItem {
  id: string;
  subject: string;
  sender: string;
  date: string;
  isUnread: boolean;
  body?: string;
  labels?: string[];
}

export interface EmailResponse {
  message: string;
  action?: 'next' | 'previous' | 'switch_tab' | 'search' | 'read' | 'mark_unread' | 'delete' | 'archive' | 'star';
  success: boolean;
  speech_response?: string;
  email?: EmailItem;
}

export interface EmailListResponse {
  emails: EmailItem[];
  speech_response: string;
  total_count?: number;
  unread_count?: number;
}

export interface VoiceCommandRequest {
  command: string;
}

export interface VoiceCommandResponse extends EmailResponse {
  recognized_command?: string;
  error_details?: string;
} 